types/mosca/index.d.ts

```diff
@@ -0,0 +1,86 @@
+// Type definitions for mosca 1.0
+// Project: https://github.com/mcollina/mosca
+// Definitions by: Joao Gabriel Gouveia <https://github.com/GabrielGouv>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export class Server {
+    opts: any;
+    modernOpts: any;
+    dedupId: any;
+    clients: any;
+    closed: boolean;
+
+    constructor(opts: any, callback?: () => void);
+
+    on(when: string, callback: (() => void) | ((client: Client) => void) | ((packet: Packet, client: Client) => void)): void;
+    once(when: string, callback: () => void): void;
+    toString(): string;
+    subscribe(topic: string, callback: () => void, done: () => void): void;
+    publish(message: Message, callback: (obj: any, packet: Packet) => void): void;
+    authenticate(client: Client, username: string, password: string,
+        callback: (obj: any, authenticated: boolean) => void): void;
+    published(packet: Packet, client: Client, callback: (obj: any) => void): void;
+    authorizePublish(client: Client, topic: string, payload: string,
+        callback: (obj: any, authorized: boolean) => void): void;
+    authorizeSubscribe(client: Client, topic: string, callback: (obj: any, authorized: boolean) => void): void;
+    authorizeForward(client: Client, packet: Packet, callback: (obj: any, authorized: boolean) => void): void;
+    storePacket(packet: Packet, callback: () => void): void;
+    deleteOfflinePacket(client: Client, messageId: number, callback: () => void): void;
+    forwardRetained(pattern: string, client: Client, callback: () => void): void;
+    restoreClientSubscriptions(client: Client, callback: () => void): void;
+    forwardOfflinePackets(client: Client, callback: () => void): void;
+    updateOfflinePacket(client: Client, originMessageId: number, packet: Packet,
+        callback: (obj: any, packet: Packet) => void): void;
+    persistClient(client: Client, callback: () => void): void;
+    close(callback: () => void): void;
```

Comment:
Callback parameter is [optional](https://github.com/mcollina/mosca/blob/v1.0.0/lib/server.js#L547).

Reply:
Fixed.